pkg/imagestore/imagestore_test.go

```diff
@@ -518,6 +523,145 @@ var _ = Context("with a data directory configured", func() {
+							ghttp.RespondWith(http.StatusOK, isoContent, isoHeader),
+						),
+					)
+
+					is, err := NewImageStore(mockEditor, dataDir, imageServiceBaseURL, false, []map[string]string{s390xVersion}, "", osImageDownloadHeadersMap, osImageDownloadQueryParamsMap, mockNmstateHandler)
+					Expect(err).NotTo(HaveOccurred())
+
+					// Pre-create the cached nmstatectl file to avoid extraction
+					nmstatectlPath, err := is.NmstatectlPathForParams(s390xVersion["openshift_version"], s390xVersion["cpu_architecture"])
+					Expect(err).NotTo(HaveOccurred())
+					Expect(os.WriteFile(nmstatectlPath, []byte("cached-nmstatectl"), 0755)).To(Succeed()) //nolint:gosec
+
+					// No CreateMinimalISOTemplate should be called for s390x
+					mockEditor.EXPECT().CreateMinimalISOTemplate(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Times(0)
+
+					Expect(is.Populate(ctx)).To(Succeed())
+
+					// Verify nmstatectl cache exists
+					_, err = os.Stat(nmstatectlPath)
+					Expect(err).NotTo(HaveOccurred())
+				})
+			})
+
+			Context("nmstatectl extraction and caching", func() {
+				DescribeTable("caches nmstatectl and creates minimal ISO for supported architectures",
+					func(arch string) {
+						archVersion := map[string]string{
+							"openshift_version": "4.18",
+							"cpu_architecture":  arch,
+							"version":           "418.84.202109241901-0",
+							"url":               ts.URL() + "/" + arch + ".iso",
+						}
+
+						// Provide valid ISO response
+						isoContent, isoHeader := isoInfo(validVolumeID)
+						ts.AppendHandlers(
+							ghttp.CombineHandlers(
+								ghttp.VerifyRequest("GET", "/"+arch+".iso"),
+								ghttp.RespondWith(http.StatusOK, isoContent, isoHeader),
+							),
+						)
+
+						is, err := NewImageStore(mockEditor, dataDir, imageServiceBaseURL, false, []map[string]string{archVersion}, "", osImageDownloadHeadersMap, osImageDownloadQueryParamsMap, mockNmstateHandler)
+						Expect(err).NotTo(HaveOccurred())
+
+						// Pre-create nmstatectl cache to avoid actual extraction
+						nmstatectlPath, err := is.NmstatectlPathForParams(archVersion["openshift_version"], archVersion["cpu_architecture"])
+						Expect(err).NotTo(HaveOccurred())
+						cacheContent := fmt.Sprintf("cached-nmstatectl-%s", arch)
+						Expect(os.WriteFile(nmstatectlPath, []byte(cacheContent), 0755)).To(Succeed()) //nolint:gosec
+
+						// Mock minimal ISO creation
+						expectedRootfs := fmt.Sprintf("http://images.example.com/boot-artifacts/rootfs?arch=%s&version=%s", arch, archVersion["openshift_version"])
+						mockEditor.EXPECT().CreateMinimalISOTemplate(gomock.Any(), expectedRootfs, arch, gomock.Any(), archVersion["openshift_version"], nmstatectlPath).Return(nil)
+
+						Expect(is.Populate(ctx)).To(Succeed())
+
+						// Verify nmstatectl cache exists
+						content, err := os.ReadFile(nmstatectlPath)
+						Expect(err).NotTo(HaveOccurred())
+						Expect(content).To(Equal([]byte(cacheContent)))
```

Comment:
Same here, we're expecting the file we wrote to be present. The only thing these are testing is that we didn't delete the file by calling `Populate`.